web/packages/teleport/src/components/ButtonLockedFeature/ButtonLockedFeature.test.tsx

```diff
@@ -75,30 +81,78 @@ describe('buttonLockedFeature', () => {
   test('it has upgrade-community href for community edition', () => {
     const version = ctx.storeUser.state.cluster.authVersion;
     cfg.isUsageBasedBilling = false;
-    renderWithContext(
-      <ButtonLockedFeature noIcon={true}>text</ButtonLockedFeature>
-    );
+    renderWithContext({
+      ui: <ButtonLockedFeature noIcon={true}>text</ButtonLockedFeature>,
+      enterprise: false,
+    });
     expect(screen.getByText('text').closest('a')).toHaveAttribute(
       'href',
       `https://goteleport.com/r/upgrade-community?${version}&utm_campaign=CTA_UNSPECIFIED`
     );
 
-    renderWithContext(
-      <ButtonLockedFeature noIcon={true} event={CtaEvent.CTA_ACCESS_REQUESTS}>
-        othertext
-      </ButtonLockedFeature>
-    );
+    renderWithContext({
+      ui: (
+        <ButtonLockedFeature noIcon={true} event={CtaEvent.CTA_ACCESS_REQUESTS}>
+          othertext
+        </ButtonLockedFeature>
+      ),
+      enterprise: false,
+    });
     expect(screen.getByText('othertext').closest('a')).toHaveAttribute(
       'href',
       `https://goteleport.com/r/upgrade-community?${version}&utm_campaign=${
         CtaEvent[CtaEvent.CTA_ACCESS_REQUESTS]
       }`
     );
   });
+
+  describe('userEventService', () => {
+    beforeEach(() => {
+      jest.spyOn(userEventService, 'captureCtaEvent');
+    });
+
+    afterEach(() => {
+      jest.resetAllMocks();
+      jest.clearAllMocks();
+    });
+
+    test('does not invoke userEventService for oss', async () => {
+      renderWithContext({
+        ui: <ButtonLockedFeature>content</ButtonLockedFeature>,
+        enterprise: false,
+      });
+
+      await userEvent.click(screen.getByText('content').closest('a'));
+      expect(userEventService.captureCtaEvent).not.toHaveBeenCalled();
+    });
+
+    test('invokes userEventService for enterprise', async () => {
+      renderWithContext({
+        ui: (
+          <ButtonLockedFeature event={CtaEvent.CTA_ACCESS_REQUESTS}>
+            content
+          </ButtonLockedFeature>
+        ),
+      });
+
+      await userEvent.click(screen.getByText('content').closest('a'));
+      expect(userEventService.captureCtaEvent).toHaveBeenCalledWith(
+        CtaEvent.CTA_ACCESS_REQUESTS
+      );
+    });
+  });
 });
 
 const ctx = createTeleportContext();
-function renderWithContext(ui: React.ReactElement) {
+
+type renderProps = {
+  ui: React.ReactElement;
+  enterprise?: boolean;
+};
+
+function renderWithContext({ ui, enterprise = true }: renderProps) {
```

Comment:
IMO an API like this would fit nicely in this case:

```ts
function renderWithContext(
  ui: React.ReactElement,
  { enterprise = true }: Options = {}
)
```
It separates the "main argument" from others that are not so important.